api/src/jvmMain/kotlin/io/kinference/compiler/serialization/JvmArraySerialization.kt

```diff
@@ -0,0 +1,68 @@
+package io.kinference.compiler.serialization
+
+import java.nio.ByteBuffer
+
+actual fun ByteArray.toShortTiledArray(blockSize: Int, blocksNum: Int): Array<ShortArray> {
+    val buffer = ByteBuffer.wrap(this).asShortBuffer()
+    return Array(blocksNum) { ShortArray(blockSize) { buffer.get() } }
+}
+
+actual fun ByteArray.toIntTiledArray(blockSize: Int, blocksNum: Int): Array<IntArray> {
+    val buffer = ByteBuffer.wrap(this).asIntBuffer()
+    return Array(blocksNum) { IntArray(blockSize) { buffer.get() } }
+}
+
+actual fun ByteArray.toLongTiledArray(blockSize: Int, blocksNum: Int): Array<LongArray> {
+    val buffer = ByteBuffer.wrap(this).asLongBuffer()
+    return Array(blocksNum) { LongArray(blockSize) { buffer.get() } }
+}
+
+actual fun ByteArray.toFloatTiledArray(blockSize: Int, blocksNum: Int): Array<FloatArray> {
+    val buffer = ByteBuffer.wrap(this).asFloatBuffer()
+    return Array(blocksNum) { FloatArray(blockSize) { buffer.get() } }
+}
+
+actual fun ByteArray.toDoubleTiledArray(blockSize: Int, blocksNum: Int): Array<DoubleArray> {
+    val buffer = ByteBuffer.wrap(this).asDoubleBuffer()
+    return Array(blocksNum) { DoubleArray(blockSize) { buffer.get() } }
+}
+
+actual fun Array<ShortArray>.toByteArray(): ByteArray {
```

Comment:
These functions look very similar. Can we reuse one common function or something like this?